crates/interpreter/src/valid.rs

```diff
@@ -764,17 +810,60 @@ impl<'a, 'm> Expr<'a, 'm> {
         Ok(())
     }
 
-    fn br_label(&self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
+    fn stitch(&mut self, branch: &SideTableBranch) {
+        let current_parser = self.parser.save().as_ptr() as isize;
+        let SideTableBranch { parser, side_table_len } = branch;
+        let entry = &mut self.side_table[*side_table_len];
+        if entry.is_none() {
+            *entry = Some(SideTableEntryView {
+                delta_ip: (current_parser - parser.as_ptr() as isize) as i32,
+                delta_stp: 0,
+                val_cnt: 0,
+                pop_cnt: 0,
+            });
+        }
+    }
+
+    fn br_label(&mut self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
+        self.push_entry(l)?;
         let l = l as usize;
         let n = self.labels.len();
-        check(l < n)?;
         let label = &self.labels[n - l - 1];
         Ok(match label.kind {
-            LabelKind::Block | LabelKind::If => label.type_.results,
+            LabelKind::Block | LabelKind::If(_) => label.type_.results,
```

Comment:
Ideally, we should only push an entry and branch in this case. For `Loop` below, we directly want to produce the correct entry without a branch.